File: 451-sort-characters-by-frequency/sort-characters-by-frequency.cpp
class Solution {
public:
 
    string frequencySort(string s) {
        ios_base::sync_with_stdio(false);
        map<char,int> mpp;
        for(char c: s){
            mpp[c]++;
        }
        vector<pair<char,int>> v(mpp.begin(),mpp.end());
        sort(v.begin(), v.end(), [](const pair<char, int>& a, const pair<char, int>& b) {
        return a.second > b.second; // Sorting based on frequency
    });
        string ans="";
        for(auto &it:v){
            ans+=string(it.second,it.first);
        }
        return ans;

    }
};

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.py
from collections import Counter

class Solution:
    def frequencySort(self, s: str) -> str:
        # Count the frequency of each character using Counter
        freq_map = Counter(s)

        # Sort the characters based on their frequency (highest first)
        sorted_chars = sorted(freq_map.items(), key=lambda x: x[1], reverse=True)

        # Construct the result string by repeating characters according to their frequency
        result = ''.join([char * freq for char, freq in sorted_chars])

        return result
